### PR TITLE
SSLRSQBR-67 Upgrade dependency on SonarQube to LTS 6.7

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: src/main/java/org/sonar/squidbridge/ProgressReport.java [hunk 1]
```diff
@@ -61,7 +60,7 @@ public void run() {
       try {
         Thread.sleep(period);
         synchronized (this) {
-          log(currentFileNumber + "/" + count + " files " + adjective + ", current file: " + currentFile.getAbsolutePath());
+          log(currentFileNumber + "/" + count + " files " + adjective + ", current file: " + currentInputFile);
```
Comment (m-g-sonar) [inline src/main/java/org/sonar/squidbridge/ProgressReport.java]: rely on the `toString` method, as described in Javadoc of inputFile.

File: src/main/java/org/sonar/squidbridge/ProgressReport.java [hunk 2]
```diff
@@ -74,9 +73,9 @@ public void run() {
     }
   }
 
-  public synchronized void start(Collection<File> files) {
-    count = files.size();
-    it = files.iterator();
+  public synchronized void start(Iterable<InputFile> inputFiles) {
```
Comment (m-g-sonar) [inline src/main/java/org/sonar/squidbridge/ProgressReport.java]: I really don't know for this and would like an advice @pynicolas .

It seems to simplify a lot things for many plugin (not for SonarJava). But it seems stupid to me to have to rely on `InputFile` (which may add external constraints to users of progressReport), while we only need them to display a path in a log. 

What do you think of only relying on an `Iterable<String> filenames` instead?

File: src/test/java/org/sonar/squidbridge/annotations/AnnotationBasedRulesDefinitionTest.java [hunk 3]
```diff
@@ -155,29 +154,26 @@ public void class_with_nosqale_annotation() throws Exception {
...
-    Repository repository = buildRepository(true, RuleClass.class);
+    Repository repository = buildRepository(RuleClass.class);
     assertThat(repository.rules()).hasSize(1);
   }
 
   @Test
   public void class_with_sqale_constant_remediation() throws Exception {
 
     @Rule(key = "key1", name = "name1", description = "description1")
-    @SqaleSubCharacteristic(SubCharacteristics.CPU_EFFICIENCY)
```
Comment (pynicolas) [inline src/test/java/org/sonar/squidbridge/annotations/AnnotationBasedRulesDefinitionTest.java]: Shouldn't we keep at least one test to assert that no subcharacteristic is saved even when we use the annotation?

Comment (m-g-sonar) [reply src/test/java/org/sonar/squidbridge/annotations/AnnotationBasedRulesDefinitionTest.java]: Added a test.

**Pull request merged**